tests/src/test/java/alluxio/worker/block/meta/SpecificTierWriteIntegrationTest.java

```diff
@@ -0,0 +1,180 @@
+              Files.createTempDir().getAbsolutePath())
+          .setProperty(PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT.format(1),
+              Files.createTempDir().getAbsolutePath())
+          .setProperty(PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT.format(1),
+              String.valueOf(CAPACITY_BYTES))
+          .setProperty(PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT.format(2),
+              Files.createTempDir().getAbsolutePath())
+          .setProperty(PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT.format(2),
+              String.valueOf(CAPACITY_BYTES)).build();
+
+  @ClassRule
+  public static ManuallyScheduleHeartbeat sManuallySchedule =
+      new ManuallyScheduleHeartbeat(HeartbeatContext.WORKER_BLOCK_SYNC);
+
+  private FileSystem mFileSystem = null;
+
+  @Before
+  public final void before() throws Exception {
+    mFileSystem = mLocalAlluxioClusterResource.get().getClient();
+  }
+
+  private void writeFileAndCheckUsage(int writeTier, long memBytes, long ssdBytes, long hddBytes)
+      throws Exception {
+    FileOutStream os = mFileSystem.createFile(
+        new AlluxioURI("/tier-" + writeTier + "_" + CommonUtils.randomAlphaNumString(5)),
+        CreateFileOptions.defaults().setWriteTier(writeTier).setWriteType(WriteType.MUST_CACHE)
+            .setLocationPolicy(new LocalFirstPolicy()));
+    os.write(BufferUtils.getIncreasingByteArray(FILE_SIZE));
+    os.close();
+
+    HeartbeatScheduler.execute(HeartbeatContext.WORKER_BLOCK_SYNC);
+
+    long totalBytes = memBytes + ssdBytes + hddBytes;
+    Assert.assertEquals("Total bytes used", totalBytes,
+        mLocalAlluxioClusterResource.get().getMaster().getInternalMaster().getBlockMaster()
+            .getUsedBytes());
+
+    Map<String, Long> bytesOnTiers =
+        mLocalAlluxioClusterResource.get().getMaster().getInternalMaster().getBlockMaster()
+            .getUsedBytesOnTiers();
+    Assert.assertEquals("MEM tier usage", memBytes, bytesOnTiers.get("MEM").longValue());
+    Assert.assertEquals("SSD tier usage", ssdBytes, bytesOnTiers.get("SSD").longValue());
+    Assert.assertEquals("HDD tier usage", hddBytes, bytesOnTiers.get("HDD").longValue());
+
+  }
+
+  private void deleteAllFiles() throws Exception {
+    List<URIStatus> files = mFileSystem.listStatus(new AlluxioURI("/"));
+    for (URIStatus file : files) {
+      mFileSystem
+          .delete(new AlluxioURI(file.getPath()), DeleteOptions.defaults().setRecursive(true));
+    }
+    // Trigger a worker heartbeat to delete the blocks.
+    HeartbeatScheduler.execute(HeartbeatContext.WORKER_BLOCK_SYNC);
+  }
+
+  @Test
+  public void topTierWrite() throws Exception {
+    writeFileAndCheckUsage(0, FILE_SIZE, 0, 0);
+    deleteAllFiles();
+    writeFileAndCheckUsage(-3, FILE_SIZE, 0, 0);
+    deleteAllFiles();
+    writeFileAndCheckUsage(-4, FILE_SIZE, 0, 0);
```

Comment:
This is essentially if the index overflows. Negative indexes are relative to the bottom, and positive indexes are relative from the top. The truncating is the intended and documented behavior.

@jsimsa any comments on this?